cargo-shuttle/src/lib.rs

```diff
@@ -112,12 +112,21 @@ impl Shuttle {
         self.run(args, provided_path_to_init).await
     }
 
-    fn find_available_port(run_args: &mut RunArgs) {
+    fn find_available_port(run_args: &mut RunArgs, services_len: usize) {
         let mut available_port = 8010;
-        for port in (available_port..=std::u16::MAX).step_by(10) {
-            if portpicker::is_free_tcp(port) && portpicker::is_free_tcp(port + 1) {
-                available_port = port;
-                break;
+        for port in (available_port..=std::u16::MAX).step_by(services_len.max(10)) {
+            if portpicker::is_free_tcp(port) {
+                let mut is_inner_ports_free = true;
+                for inner_port in port..(port + services_len as u16) {
+                    if !portpicker::is_free_tcp(inner_port) {
+                        is_inner_ports_free = false;
```

Comment:
Tip: You can use a [loop label](https://doc.rust-lang.org/reference/expressions/loop-expr.html#loop-labels) to `continue` the outer loop instead of using a boolean flag.

Reply:
I am using `is_inner_ports_free` to be sure that all the inner loop port is free. please let me know if there is a way around

Reply:
got it